app/services/analytics.rb

```diff
@@ -21,7 +21,26 @@ def track_event(event, attributes = {})
     # https://www.rubydoc.info/github/newrelic/rpm/NewRelic/Agent#add_custom_attributes-instance_method
     ::NewRelic::Agent.add_custom_attributes(
       user_id: analytics_hash[:user_id],
-      user_ip: request.remote_ip,
+      user_ip: request&.remote_ip,
+      service_provider: sp,
+      event_name: event,
+    )
+  end
+
+  def track_non_browser_event(event, attributes = {})
```

Comment:
Couple thoughts, not blocking:

- Could we infer "non browser" event by the absence of a `request`, and keep a single method?
- Or could we reuse some "base" method, where the "browser" variant decorates the `attributes` parameter before passing it down along? I guess it's made a little more complicated by the NewRelic handling as well.

Reply:
Thank you for the great suggestion. I moved back to the single `track_event` method that behaves differently based on the existence of `request`.